apf/apfElement.h

```diff
@@ -35,6 +35,7 @@ class Element
     EntityShape* getShape() {return shape;}
     FieldShape* getFieldShape() {return field->getShape();}
     void getComponents(Vector3 const& xi, double* c);
+    void getElementDofs(NewArray<double>& d);
```

Comment:
Usually, you will be using an object that inherited from `apf::ElementOf` which itself is a subclass of `apf::Element`. I guess in the case of Nedelec shapes that object would be of type `apf::MixedVectorElement`. In that case, you should be able to use `apf::ElementOf::getNodeValues()` to get a pointer to the beginning of `apf::Element::nodeData` member array, and there will be no need to add this extra member function to the `Element` class.

Reply:
Another option is to define this function as a general utility function in `apf` namespace as opposed to a member function of the `Element` class.

Reply:
@Samiullah-Malik 

We can keep this additional api here for now but I think it would be better to change the name of the api to `getElementNodeData()`. DOF is a very vague term in pumi so I would rather not use it.

Reply:
changed `getElementDofs()` to `getElementNodeData()`